scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroBucketMetadata.java

```diff
@@ -149,8 +149,10 @@ private static String validateKeyField(String keyField, Class keyClass, Schema s
           field, String.format("Key path %s does not exist in schema %s", keyPath[i], currSchema));
 
       currSchema = field.schema();
+      Schema.Type schemaTypeForUnionType = getSchemaTypeForUnionType(currSchema);
       Preconditions.checkArgument(
-          currSchema.getType() == Schema.Type.RECORD,
+          currSchema.getType() == Schema.Type.RECORD
```

Comment:
```suggestion
```

Reply:
we can omit this check because the getSchema method below will return the result of Schema#getType for non-union records :)